src/lib/converter/plugins/SourceLinkPlugin.ts

```diff
@@ -279,7 +277,7 @@ export class SourceLinkPlugin extends ConverterComponent {
         // No repository found, add path to ignored paths
         const segments = dirName.split("/");
         for (let i = segments.length; i > 0; i--) {
-            this.ignoredPaths.push(segments.slice(0, i).join("/"));
+            this.ignoredPaths.add(segments.slice(0, i).join("/"));
```

Comment:
Without a set, this could add sub-paths multiple times.

@Gerrit0 My understanding of this is that this would always ignore the entire path if a single file is ignored. Eg. if `src/components/generated` is in the `.gitignore`, this would ignore everything in `src`. I suspect that we'd rather just add the directory by itself, and not the path leading up to it.

Reply:
In other words, we might want to move this loop to the top of the function — the `dirName` should probably be checked if any part of it is ignored.

Reply:
>  this would always ignore the entire path if a single file is ignored. Eg. if `src/components/generated` is in the .gitignore, this would ignore everything in `src`.

I read this the same way, and I agree, this definitely looks wrong. I'm surprised there haven't been any bug reports about it.

Reply:
I've added a fix for this issue.